assets/aml-benchmark/components/src/prompt_crafter/main.py

```diff
@@ -0,0 +1,162 @@
+        type=str,
+        default="",
+        required=False,
+        help="The separator to be used for n-shot prompts.")
+    parser.add_argument(
+        "--prefix",
+        type=str,
+        required=False,
+        help="The prefix to be used for prompts.")
+    parser.add_argument(
+        "--label_map",
+        type=str,
+        required=False,
+        help="The label map to be used for prompts.")
+    parser.add_argument(
+        "--system_message",
+        type=str,
+        required=False,
+        help="The system message to be used for chat prompts.")
+    parser.add_argument(
+        "--output_pattern",
+        type=str,
+        required=True,
+        help="The jinja template representing the expected output to \
+         be used for few shot prompts when n_shot > 0")
+    return parser.parse_args()
+
+
+@swallow_all_exceptions(logger)
+def main(
+        test_data: str,
+        prompt_type: str,
+        n_shots: int,
+        output_pattern: str,
+        prompt_pattern: str,
+        output_file: str,
+        few_shot_separator: Optional[str] = None,
+        prefix: Optional[str] = None,
+        system_message: Optional[str] = None,
+        few_shot_data: Optional[str] = None,
+        random_seed: Optional[int] = 0,
+) -> None:
+    """Entry function for Prompt Crafter Component.
+
+    :param test_data: Path to jsonl would be used to generate prompts.
+    :param prompt_type: Type of prompt to generate.
+    :param n_shots: Number of shots to use for few-shot prompts.
+    :param output_pattern: Pattern to use for output prompts.
+    :param prompt_pattern: Pattern to use for prompts.
+    :param output_data: Path to jsonl with generated prompts.
+    :param few_shot_separator: Separator to use for few-shot prompts.
+    :param prefix: Prefix to use for prompts.
+    :param system_message: System message to use for prompts.
+    :param few_shot_data: Path to jsonl would be used to generate n-shot prompts.
+    :param random_seed: Random seed to use for prompts.
+    :return: None
+    """
+    prompt_crafter = PromptCrafter(
+        test_data=test_data,
+        few_shot_data=few_shot_data,
+        prompt_type=prompt_type,
+        n_shots=n_shots,
+        random_seed=random_seed,
+        output_pattern=output_pattern,
+        prompt_pattern=prompt_pattern,
+        few_shot_separator=few_shot_separator,
+        prefix=prefix,
+        output_file=output_file,
+        output_mltable=None,
+        metadata_keys=None,
+        label_map=None,
+        additional_payload=None,
+        system_message=system_message,
+        few_shot_pattern=None,
+    )
+    prompt_crafter.run()
+
+    log_mlflow_params(
+        prompt_type=prompt_type,
+        n_shots=n_shots,
+        prompt_pattern=prompt_pattern,
+        output_pattern=output_pattern,
+        system_message=system_message,
+        random_seed=random_seed,
+        output_dataset_checksum=resolve_io_path(output_file))
```

Comment:
nit: We can also log the checksum of test and fewshot datasets as well.

Reply:
Updated. Thanks